### PR TITLE
[#72] Add find-refs test coverage and fix its database/null-ref bugs

### PR DESCRIPTION
**Pull request by SkowronskiAndrew**
## Summary

This branch introduces a second Unity reference project used to generate deterministic test data, adds the first tests that run against that build output, and fixes the broken `find-refs` command (the original motivation, issue #72) while adding thorough test coverage for it.

### Fix `find-refs` (issue #72 + a second bug)

`find-refs` had no automated tests and two bugs that broke it against any database produced by the current `analyze`:

- **Issue #72 — database fails to open.** The connection string was hand-written using the legacy `System.Data.SQLite` `Version=3` keyword, which `Microsoft.Data.Sqlite` rejects (`Connection string keyword 'version' is not supported`). It is now built with `SqliteConnectionStringBuilder`, matching `SQLiteWriter`. The same broken string in `ExpectedDataGenerator` was fixed too.
- **Second bug — crash on ScriptableObject chains.** A ScriptableObject is a MonoBehaviour whose `m_GameObject` PPtr is `0`, so its `game_object` column holds a non-null id that matches no row; the `game_object`/`script` correlated subqueries then return `NULL` and `GetString` threw `InvalidOperationException`. Those reads are now null-checked. This broke `find-refs` for essentially any ScriptableObject reference chain.

### `--stdout` for `find-refs`

Added a `--stdout` option (mutually exclusive with `-o/--output-file`), mirroring `dump`, so chains can be written to the console. `command-find-refs.md` documents the new option and clarifies the traversal behavior: `find-refs` walks *up* the reference graph and stops at the first asset it reaches, so chains end at the immediate containing asset rather than the ultimate root.

### New `find-refs` tests

`UnityDataTool.Tests/FindRefsTests.cs` runs `analyze` on the LeadingEdge AssetBundle build in setup and covers:

- Regression for issue #72 (opens the DB without the connection-string error).
- AudioClip shared by two assets → two chains; referenced once → one chain.
- Name-only lookup matches both the MonoBehaviour and its MonoScript; adding a type narrows to one (disambiguation).
- ScriptableObject chains render `[Script = ...]` without crashing (regression for the second bug).
- Reaching `AssetBundleRoot` from a leaf asset, and confirming a target search stops at the leaf asset (documents the stop-at-first-asset traversal).
- Lookup by object id, `--find-all` (produces identical chains to the default for this data), non-existent object, empty refs table (`--skip-references`), and the `--stdout`/`-o` validator.
- Direct `refs`/`object_view` queries validating the underlying reference counts.

### Second reference Unity project (`LeadingEdge`) and reference build data

- Renamed the existing test project to `Baseline` (stable LTS versions) and added `LeadingEdge`, which tracks the newest Unity version so newer build features (Content Directory builds, serialized `Dictionary<,>`, Loadable`) can be tested. Editor scripts generate ScriptableObjects with well-known relationships between assets and AudioClips and run both AssetBundle and Content Directory builds.
- Checked in the resulting reference build output under `TestCommon/Data/LeadingEdgeBuilds` (AssetBundles, Content Directory, and build reports), used by the new tests.
- Added `AGENTS.md`/`CLAUDE.md` documentation for the projects and test-data folders.

### First test using the new build output (`dump`)

`DumpTests` gained a test that dumps the `SerializationDemo` ScriptableObject from the LeadingEdge AssetBundle build and confirms the serialized field layout and values across the core numeric types, the string, the `[SerializeReference]` managed reference, and the int array.

## Testing

`dotnet test UnityDataTool.Tests` — 231 passing, 0 failing.

### CODE REVIEW

File: ReferenceFinder/ReferenceFinderTool.cs [hunk 1]
```diff
@@ -24,21 +24,14 @@ public class ReferenceFinderTool
...
-            db = new SqliteConnection($"Data Source={databasePath};Version=3;Foreign Keys=False;");
-            db.Open();
-        }
-        catch (Exception e)
+        var db = OpenDatabase(databasePath);
+        if (db == null)
         {
-            Console.WriteLine($"Error opening database: {e.Message}");
             return 1;
         }
```
[the remark this answers is not in the document]
Comment (SkowronskiAndrew) [reply ReferenceFinder/ReferenceFinderTool.cs:36]: Done

File: ReferenceFinder/ReferenceFinderTool.cs [hunk 2]
```diff
@@ -81,33 +74,50 @@ public int FindReferences(string objectName, string objectType, string databaseP
...
-    public int FindReferences(long objectId, string databasePath, string outputFile, bool findAll)
+    public int FindReferences(long objectId, string databasePath, string outputFile, bool findAll, bool toStdout = false)
     {
         var objectIds = new List<long>();
-        SqliteConnection db;
+        var db = OpenDatabase(databasePath);
+        if (db == null)
+        {
+            return 1;
+        }
```
[the remark this answers is not in the document]
Comment (SkowronskiAndrew) [reply ReferenceFinder/ReferenceFinderTool.cs:87]: Done
